cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -335,6 +339,19 @@ func (o *objectGraph) getNodesWithTenants() []*node {
 			nodes = append(nodes, node)
 		}
 	}
+
+	return nodes
+}
+
+// getNodesWithLabels returns the list of nodes existing in the object graph with the `ClusterctlLabelName` label.
+func (o *objectGraph) getNodesWithLabels() []*node {
+	nodes := []*node{}
+	for _, node := range o.uidToNode {
+		if _, ok := node.labels[clusterctlv1.ClusterctlLabelName]; ok {
```

Comment:
I would recommend using a different label.
`clusterctl.cluster.x-k8s.io` is applied to all components managed by clusterctl, while in this use case we are explicitly focusing on objects that are not managed by clusterctl

Reply:
+1 for a more explicit label

Reply:
Added an additional label in amended commit. Let me know what you think.